integrationtest/src/de/tobiasroeser/mill/integrationtest/MillIntegrationTestModule.scala

```diff
@@ -62,26 +62,29 @@ trait MillIntegrationTestModule extends TaskModule {
     testTask(testCachedArgs)()
   }
 
+  /** Publish [[pluginsUnderTest]] and [[temporaryIvyModules]] to a local ivyPath
+    * @return tuple of
+    *         + ivyPath: Absolute path to the ivy.home directory that
+    *           [[pluginsUnderTest]] and [[temporaryIvyModules]] are published
+    *         + The content of `plugins.sc` file to be imported in `build.sc` of test projects
+    */
+  protected def publishPlugins: T[(String, String)] = T {
+    val ivyPath = T.dest / "ivyRepo"
+    val publisher = new LocalIvyPublisher(ivyPath / "local")
+    val plugins = pluginUnderTestDetails()
+    val empty = emptyJar().path
+    (plugins ++ temporaryIvyModulesDetails()).foreach {
+      case (((jar, pom), ivy), art) => publisher.publish(jar.path, empty, empty, pom.path, ivy.path, art, Nil)
+    }
+    ivyPath.toIO.getAbsolutePath -> plugins.map {
+      case (_, art) => s"import $$ivy.`${art.group}:${art.id}:${art.version}`"
+    }.mkString("\n")
+  }
+
   protected def testTask(args: Task[Seq[String]]): Task[Seq[TestCase]] = T.task {
     val ctx = T.ctx()
 
-    // publish Local
-    val ivyPath = ctx.dest / "ivyRepo"
-
-    // @return the content of `plugins.sc` file to be imported in `build.sc` of projects in [[testProjects]]
-    def publishPlugins(): String = {
-      val publisher = new LocalIvyPublisher(ivyPath / "local")
-      val plugins = pluginUnderTestDetails()
-      val empty = emptyJar().path
-      (plugins ++ temporaryIvyModulesDetails()).foreach {
-        case (((jar, pom), ivy), art) => publisher.publish(jar.path, empty, empty, pom.path, ivy.path, art, Nil)
-      }
-      plugins.map {
-        case (_, art) => s"import $$ivy.`${art.group}:${art.id}:${art.version}`"
-      }.mkString("\n")
-    }
-
-    val importFileContents = publishPlugins()
```

Comment:
The whole point of doing the local publishing inside the test task was to ensure we have a _clean_ local repo. As the repo is also used by potential publishLocal tasks of the test projects, we may end up with a modified repo which is shared with other test projects.